sensorlogger.ino

```diff
@@ -0,0 +1,850 @@
+	{ads1115phototransistor, Ads1115Pins::A0},
+	{ads1115anemometer, Ads1115Pins::A1},
+	//{voltmeter, A0}
+};
+
+// --- DS18B20 Addresses ---
+// List sensor addresses here. Get them individually with ds18b20AddressFinder.ino or similar
+// Labels should be where the sensor is located or something. Or some identifying factor.
+const DS18B20Addresses DS18B20_ADDRESSES[] = {
+		{{0x28, 0x63, 0x1B, 0x49, 0xF6, 0xAE, 0x3C, 0x88}, "MCU"}, // measuring electronics enclosure temp,has white heatshrink
+		{{0x28, 0xDC, 0xA0, 0x49, 0xF6, 0xEB, 0x3C, 0xF0}, "Batt"}, // measuring battery enclosure temp, has white heatshrink
+		//{{0x28, 0x6A, 0x64, 0x49, 0xF6, 0xBE, 0x3C, 0xF0}, "red"} // has red heatshrink
+		};
+
+uint8_t NUM_SENSORS = sizeof(SENSORS) / sizeof(SENSORS[0]);
+uint8_t NUM_DS18B20S = sizeof(DS18B20_ADDRESSES) / sizeof(DS18B20_ADDRESSES[0]);
+
+//----------------------------------------------------------
+// Other Pin selection
+// Pins D5 (SCK), D6(MISO), D7(MOSI), and D8(CS) on NodeMCU are required for the SPI for the SD card reader, so don't use them.
+// On NodeMCU, pin D1(SCL) and D2(SDA) for I2C communication
+const uint8_t CS_PIN = NodeMcuPins::D8;// cs pin for sd card
+const uint8_t LED_PIN = LED_BUILTIN_AUX; // led pin for indicator. not required.
+
+// When adding new sensors, just make a wrapper that has the following variables and methods:
+class SensorWrap
+{
+	public:
+	bool disabled = false; // disables calls to hardware
+	uint8_t num_readings = 1; // how many readings to take from this sensor (a DHT22 has 2: temperature and humidity)
+	String labels; // label(s) for the data gathered from sensor
+	virtual void init(){} // whatever needs to go into setup()
+	float getReading(uint8_t reading_num)
+	{
+		if (disabled) // if sensor init failed, don't even try to read from it from now on
+		{
+			return NAN;
+		}
+		else
+		{
+			return takeReading(reading_num);
+		}
+	} 
+	virtual float takeReading(uint8_t reading_num){return NAN;} // return measured value (NAN if no reading taken)
+};
+
+/*
+// DHT sensors
+class DhtWrap : public SensorWrap
+{
+	private:
+	DHT dht;
+
+	public:
+	DhtWrap(uint8_t pin_num, uint8_t dht_type): dht(pin_num, dht_type)
+	{
+		num_readings = 2; // 1 temp and 1 humidity
+		labels = "DHT-Temp(F), DHT-RH(%)";
+	}
+
+	void init()
+	{
+		Serial.println("DHT init");
+		dht.begin();
+	}
+
+	float takeReading(uint8_t reading_num)
+	{
+		switch (reading_num)
+		{
+			case 0:
+				return dht.readTemperature(true); // true for F, false for C
+			case 1:
+				return dht.readHumidity();
+			default:
+				return NAN;
+		}
+	}
+};
+*/
+
+
+/* ----------------
+ * DS18B20 sensor: 
+ * Reads temperature. Many identical sensors can be wired to the same pin.
+ * num_readings does not need to be set manually. It's set in init() automatically.
+ * pin_num = OneWire data pin number
+ * If using an invalid address or there is a sensor missing, it will just read -196.6F
+ */
+class DS18B20Wrap : public SensorWrap
```

Comment:
This file is pretty long and contains a lot of different classes. Maybe it would be better to structure each type of sensor into their own file which gets imported into this main file?